fda/extrapolation.py

```diff
@@ -0,0 +1,332 @@
+            derivate (numeric, optional): Order of derivative to be evaluated.
+            keepdims (bool, optional): If the image dimension is equal to 1 and
+                keepdims is True the return matrix has shape
+                nsamples x eval_points x 1 else nsamples x eval_points.
+
+        Returns:
+            (numpy.ndarray): numpy array with the evaluation of the points.
+        """
+
+        pass
+
+    def __call__(self, fdata, eval_points, derivative=0, keepdims=False):
+        """
+        Evaluate points outside the domain range. This function is a wrapper of
+        the method `extrapolate`.
+
+        Args:
+            fdata (:class:´FData´): Object where the evaluation is taken place.
+            eval_points (numpy.ndarray): Numpy array with the evalation points
+                outside the domain range. The shape of the array may be
+                `n_eval_points` x `ndim_image`, where the entry (i,j) will be
+                the coordinate j of the time i, or `nsamples` x `n_eval_points`
+                x `ndim_image`, where the first dimension indicates the sample.
+            derivate (numeric, optional): Order of derivative to be evaluated.
+            keepdims (bool, optional): If the image dimension is equal to 1 and
+                keepdims is True the return matrix has shape
+                nsamples x eval_points x 1 else nsamples x eval_points.
+
+        Returns:
+            (numpy.ndarray): numpy array with the evaluation of the points.
+        """
+
+        return self.extrapolate(fdata, eval_points, derivative=derivative,
+                                keepdims=keepdims)
+
+
+
+class PeriodicExtrapolation(Extrapolator):
+    """Extends the domain range periodically."""
+
+    def extrapolate(self, fdata, eval_points, derivative=0, keepdims=False):
+        """
+        Evaluate points outside the domain range.
+
+        Args:
+            fdata (:class:´FData´): Object where the evaluation is taken place.
+            eval_points (numpy.ndarray): Numpy array with the evalation points
+                outside the domain range. The shape of the array may be
+                `n_eval_points` x `ndim_image`, where the entry (i,j) will be
+                the coordinate j of the time i, or `nsamples` x `n_eval_points`
+                x `ndim_image`, where the first dimension indicates the sample.
+            derivate (numeric, optional): Order of derivative to be evaluated.
+            keepdims (bool, optional): If the image dimension is equal to 1 and
+                keepdims is True the return matrix has shape
+                nsamples x eval_points x 1 else nsamples x eval_points.
+
+        Returns:
+            (numpy.ndarray): numpy array with the evaluation of the points.
+        """
+
+        domain_range = fdata.domain_range
+
+
+        # Extends the domain periodically in each dimension
+        for i in range(fdata.ndim_domain):
```

Comment:
Correct me if I am mistaken. Could this loop be vectorized as:
```
((eval_points - domain_range[:, 0]) % (domain_range[:, 1] - domain_range[:, 0])) + domain_range[:, 0]
```
?

Obviously, if it is necessary to use auxiliary variables to make the code more readable is fine, but my point remains.

Reply:
Yes, you are right. Changed in e19a1f6.